### PR TITLE
add sleef/3.5.1

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

closes https://github.com/conan-io/conan-center-index/issues/4874

### CODE REVIEW

File: recipes/sleef/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_STATIC_TEST_BINS"] = False
+        self._cmake.definitions["ENABLE_LTO"] = False
+        self._cmake.definitions["BUILD_LIBM"] = True
+        self._cmake.definitions["BUILD_DFT"] = False
```
Comment (prince-chrismc) [inline recipes/sleef/all/conanfile.py:66]: I regret checking what this option requires

File: recipes/sleef/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        self._cmake.definitions["BUILD_TESTS"] = False
+        self._cmake.definitions["BUILD_INLINE_HEADERS"] = False
+        self._cmake.definitions["SLEEF_TEST_ALL_IUT"] = False
+        self._cmake.definitions["SLEEF_SHOW_CONFIG"] = True
+        self._cmake.definitions["SLEEF_SHOW_ERROR_LOG"] = False
+        self._cmake.definitions["ENFORCE_TESTER"] = False
+        self._cmake.definitions["ENFORCE_TESTER3"] = False
+        self._cmake.definitions["ENABLE_ALTDIV"] = False
+        self._cmake.definitions["ENABLE_ALTSQRT"] = False
+        self._cmake.definitions["DISABLE_FFTW"] = True
```
Comment (prince-chrismc) [inline recipes/sleef/all/conanfile.py:78]:
```suggestion
        self._cmake.definitions["DISABLE_FFTW"] = True # adds more test
```

**Review by prince-chrismc — APPROVED** [review]

**Comment by SpaceIm** [comment]
It's worth noting that sleef is the last mandatory dependency missing in CCI for libtorch.
Others like tensorpipe, xnnpack, nnpack and fbgemm are optionals (but enabled by default), we can begin libtorch recipe without them.

**Review by uilianries — APPROVED** [review]

File: recipes/sleef/all/conanfile.py (hunk 1 above)
Comment (SSE4) [reply recipes/sleef/all/conanfile.py:66]: what does it mean?

**Review by SSE4 — APPROVED** [review]

File: recipes/sleef/all/conanfile.py (hunk 1 above)
Comment (uilianries) [reply recipes/sleef/all/conanfile.py:66]: DFT is an extra library, by default it's enabled. It means discrete Fourier transform:

https://github.com/shibatch/sleef/blob/c98d92603533bae913ea35ffeab57d0441169290/src/dft/CMakeLists.txt#L332

https://sleef.org/dft.xhtml

Comment (SpaceIm) [reply recipes/sleef/all/conanfile.py:66]: It's enabled by default in upstream build files, that's true, but I had several build issues with this one. It's not required for libtorch, but someone can enable it later if needed.

**Pull request merged**